src/html.js

```diff
@@ -8,11 +8,24 @@ const escapeDictionary = {
 
 const escapeRegExp = new RegExp(
   `[${Object.keys(escapeDictionary).join("")}]`,
-  "gu",
+  "u",
 );
 
-const escapeFunction = (key) => {
-  return escapeDictionary[key];
+const escapeFunction = (string) => {
+  const stringLength = string.length;
+  let start = 0;
+  let end = 0;
+  let escaped = "";
+
+  do {
```

Comment:
no need for a  `for` loop, we can start directly as we only pass a non empty string